tests/src/flow_server_test.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { readFileSync } from 'fs';
+import * as yaml from 'yaml';
+import { retriable, runTestsForApp } from './utils.js';
+
+(async () => {
+  // TODO: Add NodeJS tests
+  // Run the tests for go test app
+  await runTestsForApp('../go/tests/test_app', 'go run main.go', async () => {
+    await testFlowServer();
+  });
+})();
+
+async function testFlowServer() {
+  const url = 'http://localhost:3400';
+  await retriable(
+    async () => {
+      const res = await fetch(`${url}/streamy`, {
+        method: 'POST',
+        body: JSON.stringify({
+          data: 1,
+        }),
+      });
+      if (res.status != 200) {
+        throw new Error(`timed out waiting for flow server to become healthy`);
+      }
+    },
+    {
+      maxRetries: 30,
+      delayMs: 1000,
+    }
+  );
+
+  const t = yaml.parse(readFileSync('flow_server_tests.yaml', 'utf8'));
+  for (const test of t.tests) {
+    console.log('path', test.path);
+    let fetchopts = {
+      method: 'POST',
+      headers: {
+        Accept: 'text/event-stream',
+        Connection: 'keep-alive',
+      },
+      body: JSON.stringify(test.post),
+    } as RequestInit;
+
+    await fetchData(`${url}/${test.path}`, fetchopts);
```

Comment:
I wonder if it would be better to use `genkit/client` library here for the test?

Reply:
Ah, good idea. Could you illuminate me on why using `genkit/client` would be better for this? Is it for compatibility reasons?

Using the raw request, we can inspect headers and response payloads to ensure the proper formatting.